src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -422,7 +422,6 @@ public static unsafe int SequenceCompareTo(ref char first, int firstLength, ref
 
         // IndexOfNullCharacter processes memory in aligned chunks, and thus it won't crash even if it accesses memory beyond the null terminator.
         // This behavior is an implementation detail of the runtime and callers outside System.Private.CoreLib must not depend on it.
-        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
```

Comment:
There are some more SpanHelpers methods marked with `AggressiveOptimization`. Delete it on all of them?

Reply:
> There are some more SpanHelpers methods marked with `AggressiveOptimization`. Delete it on all of them?

There are a few cases when because of SVM we get a non optimal codegen in R2R (similar to https://github.com/dotnet/runtime/issues/84421#issuecomment-1505287684) so I didn't want to regress SpanHelpers

Reply:
E.g. `System.SpanHelpers:LastIndexOfValueType[short,System.SpanHelpers+DontNegate`1[short]](byref,short,int)`

Reply:
So only `*IndexOfAnyValue*` (with generic math) are left with `AggressiveOptimization` in SpanHelpers

Reply:
Both R2R and Tier0 codegen for these compilated generic constructs tends to be pretty bad. I would not worry about it - we have the same problem in number of other places.

Reply:
E.g. if I remove `[AO]` from `LastIndexOfValueType` here is what I get:

```asm
; Assembly listing for method System.SpanHelpers:LastIndexOfValueType[short](byref,short,int):int
; Emitting BLENDED_CODE for X64 CPU with AVX - Windows
; ReadyToRun compilation
; optimized code
; rsp based frame
; fully interruptible
; No PGO data

G_M000_IG01:                ;; offset=0000H

G_M000_IG02:                ;; offset=0000H
       movsx    rdx, dx
       lea      rax, [(reloc 0x435488)]

G_M000_IG03:                ;; offset=000BH
       tail.jmp [rax]System.SpanHelpers:LastIndexOfValueType[short,System.SpanHelpers+DontNegate`1[short]](byref,short,int):int

; Total bytes of code 14
```
and that nested `LastIndexOfValueType` is jit-compiled. so presumably we'll get a slow Tier0 version instead of having AggressiveOpt one for start - if that is ok I can remove

Reply:
I think it is ok.

Reply:
Ok, remove them. Thus, we only have 3 uses of AO in the corelib - 1 in `AsyncTaskMethodBuilder` that says that `AO` helps it to avoid allocations in T0. And two in CastHelpers which have to be there since VM special case them to be direct calls